### PR TITLE
InfluxDB SQL: Don't show dataset dropdown

### LINKED ISSUE #78552: InfluxDB SQL: Don't display dataset dropdown

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

For InfluxDB SQL we don't need dataset dropdown to be shown. The default dataset is `iox`. 

**Who is this feature for?**

InfluxDB SQL users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/78552

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by aangelisc — APPROVED** [review]
LGTM, will wait for another review from @grafana/oss-big-tent

**Comment by gabor** [comment]
hmm.. couldn't we just rename `isPostgresInstance` to something more generic, like `showdatasetwhatever`? 

i understand that when we have `isPostgresInstance`, we can add `isFsqlInstance`, but i'd rather change this to somehow become a flag that describes a capability, not a specific database.

WDYT?

File: public/app/features/plugins/sql/components/QueryEditor.tsx
Comment (itsmylife) [reply public/app/features/plugins/sql/components/QueryEditor.tsx:17]: I think Gabor's idea will fit better and we don't need dialect in the end.

Comment (itsmylife) [reply public/app/features/plugins/sql/components/QueryEditor.tsx:17]: I tried to implement Gabor's idea but `isPostgresInstance` flag is controlling more than it should. 
So if it is a postgress instance 
- manage dataset dropdown
- decide whether the dataset should be fetched or not 

There is a feature toggle (`sqlDatasourceDatabaseSelection`) too. Which makes things a bit more complicated. 

So the current state of the component must know if the instance is postgres or not. I think a dialect will be a better fit. 

cc: @gabor

Comment (itsmylife) [reply public/app/features/plugins/sql/components/QueryEditor.tsx]: https://github.com/grafana/grafana/pull/78553/commits/6efc148fed46a57495cdd2468910e31820299b51

**Review by gabor — APPROVED** [review]
LGTM 👍 

i'd like to ask @zoltanbedi for a quick check too, he knows this codebase best, there is some interaction with the `sqlDatasourceDatabaseSelection` feature flag that is not clear to me, so it would be great if he could check this to make sure we are not making something else harder to achieve 😄

**Review by baldm0mma — APPROVED** [review]
The previous intended behavior for postgres seems to be intact. LGTM.

**Review by zoltanbedi — APPROVED** [review]
lgtm

**Pull request merged**
